synapse/logging/_remote.py

```diff
@@ -118,7 +122,7 @@ def __init__(
         if _reactor is None:
             from twisted.internet import reactor
 
-            _reactor = reactor
+            _reactor = reactor  # type: ignore[assignment]
```

Comment:
`twisted.internet.reactor` is a module that, upon import, [removes itself from `sys.modules`](
https://github.com/twisted/twisted/blob/trunk/src/twisted/internet/reactor.py#L37-L40) and calls `twisted.internet.default.install()`. This eventually [reinserts `twisted.internet.reactor` back into `sys.modules`](https://github.com/twisted/twisted/blob/d0a8ca7f28f296feefd0695acadf8dd729acfd65/src/twisted/internet/main.py#L20-L34), but now it points to a reactor instance.

Reply:
No wonder mypy is confused.

Reply:
In other spots we do:

```python
from twisted.internet import reactor as _reactor

reactor = cast(ISynapseReactor, _reactor)
```

🤷  (Could probably just put a `cast` here instead of an ignore to be consistent.)